lib/browser/api/web-contents.js

```diff
@@ -206,60 +207,135 @@ WebContents.prototype.executeJavaScriptInIsolatedWorld = async function (code, h
 
 // Translate the options of printToPDF.
 WebContents.prototype.printToPDF = function (options) {
-  const printingSetting = {
+  const printSettings = {
     ...defaultPrintingSetting,
     requestID: getNextId()
   }
+
   if (options.landscape) {
-    printingSetting.landscape = options.landscape
-  }
-  if (options.fitToPageEnabled) {
-    printingSetting.fitToPageEnabled = options.fitToPageEnabled
+    if (typeof options.landscape !== 'boolean') {
+      const error = new Error('landscape must be a Boolean')
+      return Promise.reject(error)
+    }
+    printSettings.landscape = options.landscape
   }
+
   if (options.scaleFactor) {
-    printingSetting.scaleFactor = options.scaleFactor
+    if (typeof options.scaleFactor !== 'number') {
+      const error = new Error('scaleFactor must be a Number')
+      return Promise.reject(error)
+    }
+    printSettings.scaleFactor = options.scaleFactor
```

Comment:
This is brainstorming a little bit on how to reduce the copy-paste effect introduced by the improved type checks. The code seems like it would benefit from extract-method, what do you think about something like this?

```js

const parameterTypes = {
  marginsType: 'number',
  printSelectionOnly: 'boolean',
  scaleFactor: 'number',
};

for (const [key, expectedType] of Object.entries(parameterTypes)) {
    const val = options[key];
    const type = typeof val;
    if (type === expectedType) {
      printSettings[key] = val;
    } else if (type !== 'undefined') {
      const err = new Error(`${key} must be a ${expectedType}`);
      return Promise.reject(error);
    }
}
```

Doesn't have to be this verbatim -- for example if it was a `copyTypedProperty()` function instead then it could be applied to `options.pageRanges` as well as `options` -- but whatever approach, it would be good to consider reducing code duplication here

Reply:
no - it's `pageRange` on the Chromium side but i set it to `pageRanges` for parity with `webContents.print`

Reply:
i don't disagree, but i think we should also consider that most of these require edge case or nested type checking, and at that point we already can only apply this deduplication to ~4 parameters of the 8. At that point i'd start to become concerned that the abstraction would perhaps hamper readability 🤔

Reply:
Yeah, that's true. OK.